baseclasses/BaseSolver.py

```diff
@@ -116,7 +163,7 @@ def getOption(self, name):
            Return the current value of the option.
         """
 
-        if name in self.defaultOptions:
+        if name in self.defaultOptions or not self.checkDefaultOptions:
             return self.options[name]
```

Comment:
Since its now possible to bypass all checks in `setOption`, this call `return self.options[name]` may fail for an option that has not been added. 
Do we want to add a check e.g., `if name in self.options`, check hereto make sure it exists and if not, then raise an Error `Option {name} was not found. Make sure the option has been added.` or something similar.
We also need to add test for this `getOption` call.